sdk/js/executor.js

```diff
@@ -0,0 +1,300 @@
+
+export class Executor {
+    constructor (cmgr, account, ms) {
+        this.cmgr = cmgr
+        this.account = account
+        this.ms = ms
+        this.gpacts = new Map()
+    }
+
+    registerGPACT(chainID, gpactAddr) {
+        this.gpacts.set(chainID, gpactAddr)
+    }
+
+    async crosschainCall(root) {
+        // Generate a random transaction id
+        var temp = '0b'
+        for (let i = 0; i < 256; i++) {
+            temp += Math.round(Math.random())
+        }
+        var transID = BigInt(temp)
+
+        // Start event
+        console.log("start...")
+        var [startEvent, startEventSig, err] = await this.start(transID, root)
+        if (err != null) {
```

Comment:
in the event contacting the message store fails inside `start()`, it appears that `err` will be null (see comment further down), even though the routine didn't fully succeed. Perhaps we can set `err` in those conditions inside the `start()` funciton, or check whether `startEventSign` is null here to detect whether the start routine was really successful?